runtime/tests/clinkerffi/downcall.c

```diff
@@ -4525,3 +4525,147 @@ addIntAndShortsFromUnion_Nested4ShortStruct(int arg1, union_Int_NestedStruct_4_S
 			+ arg2.elem2.elem2 + arg2.elem2.elem3 + + arg2.elem2.elem4;
 	return intSum;
 }
+
+/**
+ * Get a new struct by adding each boolean element of three structs
+ * (dereferenced from pointers) with two booleans.
+ *
+ * @param arg1 a pointer to on-heap struct with two booleans
+ * @param arg2 a pointer to off-heap struct with two booleans
+ * @param arg3 a pointer to on-heap struct with two booleans
```

Comment:
Why is it relevant here whether the parameters point into the heap or not?

Reply:
We need to check whether each heap argument is correctly accessed in our code by mixing with the native  & heap memory arguments in tests.